pkg/registries/filter.go

```diff
@@ -0,0 +1,227 @@
+package registries
+
+import (
+	"regexp"
+	"sync"
+)
+
+type filterMode uint8
+
+const (
+	filterModeBoth filterMode = iota
+	filterModeWhite
+	filterModeBlack
+	filterModeNone
+)
+
+type failedRegexp struct {
```

Comment:
I actually am not a huge fan of this, but I could be swayed. @shawn-hurley what's most golang-like, in your opinion?